skbio/stats/composition.py

```diff
@@ -1701,3 +1711,283 @@ def _check_orthogonality(basis):
     basis = np.atleast_2d(basis)
     if not np.allclose(basis @ basis.T, np.identity(len(basis)), rtol=1e-4, atol=1e-6):
         raise ValueError("Basis is not orthonormal")
+
+
+def _welch_ttest(x1, x2):
+    # See https://stats.stackexchange.com/a/475345
+    n1 = x1.size
+    n2 = x2.size
+    m1 = np.mean(x1)
+    m2 = np.mean(x2)
+
+    v1 = np.var(x1, ddof=1)
+    v2 = np.var(x2, ddof=1)
+
+    pooled_se = np.sqrt(v1 / n1 + v2 / n2)
+    delta = m1 - m2
+
+    tstat = delta / pooled_se
+    df = (v1 / n1 + v2 / n2) ** 2 / (
+        v1**2 / (n1**2 * (n1 - 1)) + v2**2 / (n2**2 * (n2 - 1))
+    )
+
+    # two side t-test
+    p = 2 * t.cdf(-abs(tstat), df)
+
+    # upper and lower bounds
+    lb = delta - t.ppf(0.975, df) * pooled_se
+    ub = delta + t.ppf(0.975, df) * pooled_se
+    return pd.DataFrame(
+        np.array([tstat, df, p, delta, lb, ub]).reshape(1, -1),
+        columns=["T statistic", "df", "pvalue", "Difference", "CI(2.5)", "CI(97.5)"],
+    )
+
+
+@experimental(as_of="0.5.9")
+def dirmult_ttest(
+    table: pd.DataFrame,
+    grouping: str,
+    treatment: str,
+    reference: str,
+    pseudocount: float = 0.5,
+    draws: int = 128,
+    seed=0,
```

Comment:
The default value for `seed` can be `None`. See [numpy doc](https://numpy.org/doc/stable/reference/random/generator.html). See also skbio function [`get_rng`](https://github.com/scikit-bio/scikit-bio/blob/72104932830ace439ed64c298ce354264e23c75c/skbio/util/_misc.py#L217).